text/0000-remove-stacks-mixins.md

```diff
@@ -0,0 +1,202 @@
+
+Target ID is an identifier specified on the runtime base image that must be provided to buildpacks as `CNB_TARGET_ID` during the build process.
+This allows buildpacks to change their behavior if a run image is selected (e.g., distroless) that has special properties outside of OS, architecture, etc.
+
+For Linux-based images, each field should be canonicalized against values specified in `/etc/os-release` (`$ID` and `$VERSION_ID`).
+The `os.version` field in an base image `config` may contain combined distribution and version information, but it is not used by the lifecycle.
+
+For Windows-based images, Distribution should be empty. Version should be the [suggested value of `os.version`](https://github.com/opencontainers/image-spec/blob/main/config.md#properties) in the OCI spec (e.g., `10.0.14393.1066`).
+
+The `stacks` list in `buildpack.toml` is replaced by a `targets` list, where each entry corresponds to a different buildpack image that is exported into a [manifest index](https://github.com/opencontainers/image-spec/blob/master/image-index.md).
+Each entry may contain multiple valid values for Distribution and/or Version, but only a single OS, Architecture, and Variant.
+If the `targets` list is empty and `/bin/build` is present, a target with `os = "linux"` and `arch = "x86_64"` is assumed by tools reading `buildpack.toml`.
+If the `targets` list is empty and `/bin/build.bat` or `/bin/build.exe` is present, a target with `os = "windows"` and `arch = "x86_64"` is assumed by tools reading `buildpack.toml`.
+
+App image builds fail if the build image and selected run image have mismatched metadata. We may introduce flags or additional labels to skip this validation (e.g., for cross-compilation or minimal runtime base images).
+An image without a specified Distribution is compatible with images specifying any Distribution.
+An image specifying a Distribution without a Version is compatible with images specifying any Versions of that Distribution.
+
+When an app image is rebased, `rebaser` must fail if the new run image and previous run image have mismatched metadata. This check can be skipped for Distribution and Version by passing a new `--force` flag to `rebaser`.
+
+#### Example: buildpack.toml `targets` table
+
+```toml
+[[targets]]
+os = "linux"
+arch = "x86_64"
+[[targets.distributions]]
+name = "ubuntu"
+versions = ["18.04", "20.04"]
+
+[[targets]]
+os = "linux"
+arch = "x86_64"
+[[targets.distributions]]
+name = "ubuntu"
+versions = ["14.04", "16.04"]
+
+[[targets]]
+os = "linux"
+arch = "arm"
+variant = "v6"
+[[targets.distributions]]
+name = "ubuntu"
+versions = ["14.04", "16.04"]
+```
+
+## Runtime Metadata
+
+To allow different runtime base images to be used, and to support cross-compilation in the future, buildpacks may need access to the runtime base image's target metadata.
+The following environment variables will be available to buildpacks directly in the build-time environment (not via `/platform/env`):
```

Comment:
This part is a little unclear to me. This appears to be talking about a set of environment variables that are given to the buildpacks during **build-time**. 

1. It's not talking about these environment variable being set on the run image right?
2. What determines what these environment variables should be set to?
3. What does happen if the platform provides these environment variables via `/platform/env` as a way to configure the desired outcome? Is it explicitly something we don't want to allow?

Reply:
1. Correct: "directly in the build-time environment"
2. "runtime base image's target metadata" -- same metadata as above, but for the runtime base image
3. All `CNB_` env vars are set directly so that they are unaffected by "strict" mode that forces buildpacks to read the env vars from `/platform/env`